metagpt/provider/openai_api.py

```diff
@@ -323,3 +325,28 @@ async def amoderation(self, content: Union[str, list[str]]):
     async def _amoderation(self, content: Union[str, list[str]]):
         rsp = await self.llm.Moderation.acreate(input=content)
         return rsp
+
+    @staticmethod
+    def __get_rpm(config: Config) -> int:
+        session_key = config.get("OPENAI_SESSION_KEY", "")
+        default_rpm = int(config.get("RPM", 10))
+        if len(session_key) > 0:
+            try:
+                response = requests.get(
+                                        "https://api.openai.com/dashboard/rate_limits",
+                                        headers={'Authorization': f'Bearer {session_key}'},
+                                        timeout=10,
+                                        proxies={'https': openai.proxy}
+                                        )
+            except Exception as e:
+                logger.error(f"Connection to api.openai.com failed:{e}.Setting rpm to default parameter.")
+                return default_rpm
+            
+            if response.status_code == 200:
+                limit_dict = json.loads(response.text)[config.openai_api_model]
```

Comment:
what if the user configured openai_api_model value not in the response's json

Reply:
> what if the user configured openai_api_model value not in the response's json

This response's json contains all of the models supported by openai. So if the user put openai_api_model value which not in keys of response, it means that user put a wrong model name and it should raise a exception.